### ISSUE TITLE
梯度回传不到word embedding层

### ISSUE THREAD

**Issue by zhiqix** [THUDM/ChatGLM3#1117]
### System Info / 系統信息

datasets==2.14.4
matplotlib==3.7.2
nltk==3.8.1
numpy==1.24.1
pandas==2.0.3
sentence_transformers==2.2.2
torch==2.0.1+cu118
tqdm==4.66.1
transformers==4.33.1

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [X] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

根据模型的输出的logits进行梯度回传
probs = torch.softmax(self._model(input_ids=ids.to(self._device))["logits"], -1)

### Expected behavior / 期待表现

以下为回传后的梯度信息，可以看到，除了embedding中的梯度为0，其余layers的所有参数均有梯度。
transformer.embedding.word_embeddings.weight
tensor([[0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.],
        ...,
        [0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.]], device='cuda:0')

transformer.encoder.layers.0.input_layernorm.weight
tensor([-4.5584e-11,  3.8643e-11,  1.6178e-10,  ...,  2.2602e-11,
        -4.1157e-11, -3.0692e-10], device='cuda:0')

transformer.encoder.layers.0.self_attention.query_key_value.weight
tensor([[ 8.9492e-15, -4.8874e-16, -1.7847e-13,  ..., -7.3466e-14,
          2.4229e-14, -6.1280e-14],
        [ 7.5851e-15, -2.8725e-16, -6.9896e-14,  ..., -6.6415e-14,
         -4.3015e-16,  3.1866e-13],
        [-3.8896e-14,  1.9012e-15,  8.4087e-14,  ...,  9.0414e-14,
         -3.4556e-15,  6.2374e-14],
        ...,
        [-3.9253e-13, -4.7591e-14, -5.5059e-12,  ...,  4.7847e-12,
         -8.2938e-14,  1.6261e-12],
        [-3.8606e-13, -4.3409e-14, -1.0324e-11,  ...,  7.9638e-12,
          2.9511e-13,  8.7555e-13],
        [-1.1991e-12, -3.1184e-15, -9.2819e-12,  ...,  1.2586e-11,
          1.7464e-13, -4.6796e-13]], device='cuda:0')

**Comment by zRzRzRzRzRzRzR** [comment]
word embedding本来就没有训练呀

**Comment by zhiqix** [comment]
> word embedding本来就没有训练呀

收到，感谢感谢！
